### PR TITLE
perf: use no hash hasher to perf hashmap on yamux and tentacle

### PR DESCRIPTION
**Pull request by driftluo**

### CODE REVIEW

**Review by TheWaWaR — APPROVED** [review]

File: tentacle/src/service.rs [hunk 1]
```diff
@@ -724,7 +720,8 @@ where
         let handles = self.session_handles_open(self.next_session);
 
         let mut by_name = HashMap::with_capacity(self.protocol_configs.len());
-        let mut by_id = HashMap::with_capacity(self.protocol_configs.len());
+        let mut by_id =
```
Comment (quake) [inline tentacle/src/service.rs:723]: unnecessary change?

**Review by quake — APPROVED** [review]

File: tentacle/src/service.rs (hunk 1 above)
Comment (driftluo) [reply tentacle/src/service.rs:723]: https://doc.rust-lang.org/nightly/std/collections/struct.HashMap.html#method.with_capacity

`with_capacity` only impl for `RandomState` hasher
